services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/extensions/VersionedRecordExtension.java

```diff
@@ -101,39 +145,12 @@ public WriteModification beforeWrite(DynamoDbExtensionContext.BeforeWrite contex
             return WriteModification.builder().build();
         }
 
-        Map<String, AttributeValue> itemToTransform = new HashMap<>(context.items());
-
-        String attributeKeyRef = keyRef(versionAttributeKey.get());
-        AttributeValue newVersionValue;
-        Expression condition;
-        Optional<AttributeValue> existingVersionValue =
-            Optional.ofNullable(itemToTransform.get(versionAttributeKey.get()));
-
-        if (!existingVersionValue.isPresent() || isNullAttributeValue(existingVersionValue.get())) {
-            // First version of the record
-            newVersionValue = AttributeValue.builder().n("1").build();
-            condition = Expression.builder()
-                                  .expression(String.format("attribute_not_exists(%s)", attributeKeyRef))
-                                  .expressionNames(Collections.singletonMap(attributeKeyRef, versionAttributeKey.get()))
-                                  .build();
-        } else {
-            // Existing record, increment version
-            if (existingVersionValue.get().n() == null) {
-                // In this case a non-null version attribute is present, but it's not an N
-                throw new IllegalArgumentException("Version attribute appears to be the wrong type. N is required.");
-            }
+        Pair<AttributeValue, Expression> updates = getRecordUpdates(versionAttributeKey.get(), context);
```

Comment:
Refactoring (extracting code to a separate method) is great, but it could make it a bit hard to review the new change. Can we remove the refactoring from this commit and add it to a separate commit, if possible so that we can review the new change?

Reply:
The refactoring here was done by the person who  opened the [original PR](https://github.com/aws/aws-sdk-java-v2/pull/5565/files). I just ported their changes over.
I’m going to end up reverting those changes because IMO it makes it harder to read.